packages/sources/universal-v3/src/utils/LambdaController.ts

```diff
@@ -0,0 +1,348 @@
+      },
+      FunctionName,
+      Handler: 'index.handler',
+      Role: this.config.LAMBDA_ROLE_ARN,
+      Runtime: 'nodejs16.x',
+      Description: 'Universal adapter sandbox',
+      Timeout: Math.round(this.config.LAMBDA_SANDBOX_TIMEOUT_MS / 1000),
+      MemorySize: this.config.LAMBDA_MEMORY_SIZE_MB,
+      Environment: {
+        Variables: {
+          LOG_LEVEL: this.config.LAMBDA_LOG_LEVEL,
+        },
+      },
+      // VpcConfig: // This setting allows for putting the Lambda behind a VPC
+    }
+
+    return new Promise((resolve, reject) => {
+      this.lambda.createFunction(params, (err, data) => {
+        if (err) {
+          this.handleLambdaErrorResponse(
+            err,
+            retryCount,
+            FunctionName,
+            sandboxRequestData,
+            resolve,
+            reject,
+          )
+          return
+        }
+
+        this.logger.debug(`Lambda creation response: ${JSON.stringify(data)}`)
+
+        // Disabling no-non-null-assert since executeRequestInLambda already ensures initalization
+        /* eslint-disable @typescript-eslint/no-non-null-assertion */
+        this.deployedLambdaFunctions!.timeLastUsed[FunctionName] = Date.now()
+        this.deployedLambdaFunctions!.count++
+        /* eslint-enable @typescript-eslint/no-non-null-assertion */
+
+        setTimeout(
+          () => resolve(this.executeRequestInLambda(FunctionName, sandboxRequestData, retryCount)),
+          this.config.LAMBDA_INIT_TIME_MS,
+        )
+      })
+    })
+  }
+
+  // This recursive function returns all deployed Lambda functions as a Record from the name to
+  // the time the function was last used.  The TimeLastUsed is initally set to zero.
+  // Documentation on lambda.listFunctions: https://docs.aws.amazon.com/lambda/latest/dg/API_ListFunctions.html
+  private getDeployedLambdaFunctions = async (
+    Marker?: string,
+  ): Promise<DeployedLambdaFunctions> => {
+    return new Promise((resolve) => {
+      this.lambda.listFunctions({ Marker }, async (err, data) => {
+        if (err) {
+          this.logger.error(`Unexpected Lambda listFunctions error: ${err}`)
+          throw Error(`Error fetching list of deployed Lambda functions: ${err}`)
+        }
+
+        const functions = data.Functions
+
+        if (!functions) {
+          resolve({ count: 0, timeLastUsed: {} })
+          return
+        }
+
+        let deployedLambdaFunctions: DeployedLambdaFunctions = { count: 0, timeLastUsed: {} }
+
+        // Each call to getDeployedLambdaFunctions fetches up to 50 Lambda functions at a time.
+        // If more than 50 Lambda functions exist, the next page must be fetched using the NextMarker.
+        if (data.NextMarker) {
+          deployedLambdaFunctions = await this.getDeployedLambdaFunctions(data.NextMarker)
+        }
+
+        functions.forEach(({ FunctionName }) => {
+          if (FunctionName) deployedLambdaFunctions.timeLastUsed[FunctionName] = 0
```

Comment:
Would it be better to use something from the `listFunctions` response instead of 0? Such as:
```
         "LastModified": "string",
         "LastUpdateStatus": "string",
```

Reply:
hmmmm that's a good call.

On first initialization, we should probably keep the OLDEST lambda function as that is the one that is most likely to be most frequently used.  Then, after the initialization, we would keep the existing logic which deletes the least-recently-used function.

I'll put some more thought into this.

Reply:
`LastUpdateStatus` will always be null for our use case since functions are never updated after being created (I have tested this).  They are only either created or deleted.  Additionally, `LastModified` will always equal the time of creation.

As for a solution to the initial ordering:

Imagine we have deployed Lambda functions `0` through `n`, in sorted order, with `0` being the oldest function and `n` being the newest.

We assign the `timeLastUsed` in reverse order, such that the oldest function is assigned `n` and the newest is assigned `0`.  This means that, as soon as the max # of functions is exceeded, the newest deployed function is first to be deleted.

However, since `n` will always be less than the current timestamp, if a new function is created _after_ initialization, we maintain the functionality where the least-recently-used function is deleted first.

Does that make sense?